### PR TITLE
[Lang] Add ti.serialize and ti.loop_config

### PR DESCRIPTION
**Pull request by lin-hitonami**
Related issue = #4530

### CODE REVIEW

File: python/taichi/lang/misc.py [hunk 1]
```diff
@@ -510,19 +510,36 @@ def loop_unique(val, covers=None):
...
+        get_runtime().prog.current_ast_builder().strictly_serialize()
 
 
+parallelize = _parallelize
+
 serialize = lambda: parallelize(1)
 
 
-def block_dim(dim):
+def _block_dim(dim):
```
Comment (k-ye) [inline python/taichi/lang/misc.py:524]: Let's not yet remove this functions, just mark them as deprecated?

File: taichi/transforms/lower_ast.cpp [hunk 2]
```diff
@@ -204,9 +204,8 @@ class LowerAST : public IRVisitor {
       auto end = stmt->end;
       flatten_rvalue(begin, &fctx);
       flatten_rvalue(end, &fctx);
-      bool is_good_range_for =
-          current_block_depth_ == 1 || detected_fors_with_break_.find(stmt) ==
-                                           detected_fors_with_break_.end();
+      bool is_good_range_for = detected_fors_with_break_.find(stmt) ==
```
Comment (strongoier) [inline taichi/transforms/lower_ast.cpp:207]: Why this change? I forget what is the behavior of top-level break now...

Comment (lin-hitonami) [reply taichi/transforms/lower_ast.cpp:207]: We are now supporting serial fors at the outermost scope, and they can contain break statements, so we should convert them to while statements.

**Review by strongoier — APPROVED** [review]
LGTM!

File: python/taichi/lang/misc.py [hunk 3]
```diff
@@ -510,19 +510,32 @@ def loop_unique(val, covers=None):
...
     """Set the number of threads in a block to `dim`.
     """
     get_runtime().prog.current_ast_builder().block_dim(dim)
 
 
+def loop_config(block_dim=None, serialize=None, parallelize=None):
+    if block_dim is not None:
```
Comment (ailzhang) [inline python/taichi/lang/misc.py:530]: nit: add a docstring since it's user facing?

File: python/taichi/lang/misc.py [hunk 4]
```diff
@@ -510,19 +510,32 @@ def loop_unique(val, covers=None):
...
+        _block_dim(block_dim)
+
+    if serialize:
```
Comment (ailzhang) [inline python/taichi/lang/misc.py:533]: Looks like serialize should be a boolean defaulting to false instead of None?

**Review by ailzhang — APPROVED** [review]
LGTM!
